### DISCUSSION TITLE
Invalidate with Cache Tags

### DISCUSSION THREAD

**Discussion by MokDevelopment** [vercel/next.js#41183] [Help]
It would be wonderful to use cache tags instead of having to invalidate every page cache individually by url when using ISR. 

**Example would be:**  I have blog article pages, that invalidate their Next.js caches when saved on the cms... cool already... 

... but all the other articles + overviews and some other pages really should invalidate their caches as well .. why? because I use a teaser grid on all of these pages, that include articles as teasers. 

Instead of invalidating by route / url of one specific article there really should be "cache tags" like in other frameworks. 

I would like to do things like to mark or tag pages with "articles:all" or "articles:en" or "articles:es" or "article:45"
to have some logic behind. In this moment the detail pages of the article are up to date ... but my overviews or every other content that holds teasers to the before changed item are constantly not up to date. 

Does this concept exist already in Next.js?
Are there work arounds?

Thanks

**Comment by lucasconstantino** [comment]
I've just published an initial package for this purpose: [next-cache-tags](https://github.com/lucasconstantino/next-cache-tags). Although quite a naive solution, I believe it's a good enough foundation of how such solution could work.
